provider/datasource_application_dashboard_definition.go

```diff
@@ -157,29 +157,32 @@ func (a *ApplicationDashboardDefinitionDataSource) Read(ctx context.Context, req
 		db.AddStreamConsumer(consumer)
 	}
 
-	if kinesisData, err = a.gatherKinesisData(ctx, metadata.Cloud.Aws.Kinesis); err != nil {
-		response.Diagnostics.AddError(fmt.Sprintf("can not get kinesis data"), err.Error())
-		return
-	}
-
 	for _, kinsumer := range metadata.Cloud.Aws.Kinesis.Kinsumers {
-		db.AddCloudAwsKinesisKinsumer(kinsumer.StreamNameFull, kinesisData[kinsumer.StreamNameFull])
+		db.AddCloudAwsKinesisKinsumer(kinsumer)
 	}
 
 	for _, producer := range metadata.Stream.Producers {
 		if !producer.DaemonEnabled {
 			continue
 		}
 
-		db.AddStreamProducerDaemon(producer.Name)
+		db.AddStreamProducerDaemon(producer)
 	}
 
 	for _, writer := range metadata.Cloud.Aws.Kinesis.RecordWriters {
-		db.AddCloudAwsKinesisRecordWriter(writer.StreamName, kinesisData[writer.StreamName])
+		db.AddCloudAwsKinesisRecordWriter(writer)
+	}
+
```

Comment:
just do this instead?
```golang
	for _, stream := range metadata.Cloud.Aws.Kinesis.Kinsumers {
		db.AddCloudAwsKinesisStream(stream)
	}
	for _, stream := range metadata.Cloud.Aws.Kinesis.RecordWriters {
		db.AddCloudAwsKinesisStream(stream)
	}
```

Reply:
this ref via github looks strange, but i see your point and will adjust it